### DISCUSSION TITLE
Обновление приложения у пользователя.

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#455] [Как это сделать?] [answered]
Было замечено что, если внести изменение в приложение ( изменить какой-либо код ), то у пользователя эти изменения почему то не применяются сразу. Под инкогнито всё применяется сразу, а если пользователь продолжает работать в браузере под своей учёткой, то не всегда. Есть какие-то решения для выхода из этой ситуации?

**Comment by poluninatm** [comment, marked as answer]
Какие именно коды вы имеете ввиду?
Опишите подробнее ситуацию.

Основные типы кодов работают на сервере и меняются сразу, как только вы
сохранили изменения. Если таблица (например в цикле) не пересчитывалась
после изменения кодов, изменений на клиенте, который держит ее открытой и
не будет.

Ширина поля является величиной настраиваемой на клиенте, поэтому
подцепляется при изменении только, если изменить актуальность полей для
таблицы.


Ср, 18 окт. 2023 г. в 07:34, vldekov ***@***.***>:

**Reply by vldekov** [reply]
Например, был изменен код действия при добавлении для поля РегНомер. Но почему то у пользователя он применился обновленный только под инкогнито. Может быть это связано с кэшом браузера?

**Comment by poluninatm** [comment]
Код действия, в том числе при добавлении, обрабатывается на сервере. Сервер
ничего не знает про кеши браузера.

Что должен был сделать этот код действия?


Ср, 18 окт. 2023 г. в 07:50, vldekov ***@***.***>:

**Reply by vldekov** [reply]
ad0=:set(table: $#ntn; field: 'regnomer' = $str1; where: 'regnomer' = ""; log: true)
str1:str`$fix1+"_"+$fix4`
fix1:dateFormat(date: $#nd; format: "Y")
fix4:listCount(list: $sel1)+2
sel1:selectList(table: $#ntn; field: 'regnomer')

**Comment by poluninatm** [comment]
Должно работать при любом типе пользователя.
Но если у вас в таблице была другая строка с пустым регномером - могла
обновиться она, лучше привязываться к ид добавляемой строки.

Также в конструкциях типа listCount из listSelect лучше брать поле id - для
простых таблиц это менее ресурсоемко.

Ср, 18 окт. 2023 г. в 08:04, vldekov ***@***.***>:

**Reply by vldekov** [reply]
Если пользователь открыл панель для заполнения новой записи, а пока он её заполнял в это время на сервере количество записей обновилось другим пользователем, то после нажатия кнопки сохранить этот код действия посчитает количество всех записей ? Как исключить задвоения значений в поле РегНомер?

**Comment by poluninatm** [comment]
Да, время открытия панели не имеет значения. Код посчитает количество всех
записей на момент выполнения кода.

Если это простая таблица и в перспективе в ней будет много записей - это
ресурсоемкая операция.

Ср, 18 окт. 2023 г. в 08:42, vldekov ***@***.***>:
